### PR TITLE
Integrate kind_provisioner.sh based setup for prow testing jobs (both single and multicluster jobs)

### PR DESCRIPTION
**Pull request by su225**
~~**DO NOT MERGE** - This is a PR to make sure that kind setup and tests in prow are working fine. Currently it is for experimentation purposes only. Once all issues are fixed, we will merge it.~~

~~**NOTE:** As part of experimentation, I have changed files which should ideally be merged to istio/common-files and pulled here. Once changes are finalized, it will be merged to common-files and updated in istio/istio and istio/istio.io repositories.~~

It is now ready to merge. Multicluster topology (both network and istio control plane) is externalized to a JSON file and passed to `setup_kind_clusters` in `kind_provisioner.sh` in common-files repo. 

[X] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[X] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: common/scripts/kind_provisioner.sh [hunk 1]
```diff
@@ -98,42 +98,54 @@ function cleanup_kind_cluster() {
...
+  # explicitly disable shellcheck since we actually want $NAME to expand now
+  # shellcheck disable=SC2064
+  trap "cleanup_kind_cluster ${NAME}" EXIT
+
+    # If config not explicitly set, then use defaults
+  if [[ -z "${CONFIG}" ]]; then
+      # Kubernetes 1.15+
+      CONFIG=${DEFAULT_CLUSTER_YAML}
+      # Configure the cluster IP Family only for default configs
+    if [ "${IP_FAMILY}" = "ipv6" ]; then
```
Comment (nmittler) [inline common/scripts/kind_provisioner.sh:138]: Should this be done even if `CONFIG` is specified?

Comment (su225) [reply common/scripts/kind_provisioner.sh:138]: Same as in master branch - https://github.com/istio/istio/blob/master/prow/lib.sh#L155
I had changed, but put that back after I had issues with `ipv6` cluster setup. Not sure, why it was like that. Maybe the rationale is "if cluster configuration is explicitly given then all these issues are already taken care of"?

**Review by nmittler — APPROVED** [review]

**Comment by irisdingbj** [comment]
Can we provide a way to let prow job or users to sepcify `CLUSTER_TOPOLOGY_CONFIG_FILE`? Now we are using a fixed one from `prow/config/topology/multicluster.json`

We need this as in some cases we need different cluster settings.  @nmittler @su225

**Comment by su225** [comment]
> Can we provide a way to let prow job or users to sepcify `CLUSTER_TOPOLOGY_CONFIG_FILE`? Now we are using a fixed one from `prow/config/topology/multicluster.json`
> 
> We need this as in some cases we need different cluster settings. @nmittler @su225

@irisdingbj - Probably, we can edit `integ-suite-kind.sh` to pass `CLUSTER_TOPOLOGY_CONFIG_FILE` as an environment variable. Kind provisioner should be fine for multicluster configurations as long as it is not IPv6. 

@nmittler - Do we need to make this flexible as suggested or can we keep this for another time?

**Comment by nmittler** [comment]
@su225 

> @nmittler - Do we need to make this flexible as suggested or can we keep this for another time?

Would it be hard to just add this now? Seems like you've already done most of the heavy lifting, no? This would unblock @irisdingbj

**Comment by su225** [comment]
@irisdingbj - Added `--topology-config` parameter so that you can specify your own configuration.

**Comment by irisdingbj** [comment]
@su225 @nmittler Thanks for offering this function!

**Pull request merged**